src/models/job/index.js

```diff
@@ -17,6 +17,6 @@ export default Object.freeze({
   pipeline: PropTypes.string.isRequired,
   started: PropTypes.instanceOf(Date),
   status: PropTypes.oneOf(Object.values(ProgressStatus)).isRequired,
-  steps: PropTypes.arrayOf(PropTypes.exact(StepModel)),
+  steps: PropTypes.arrayOf(PropTypes.exact(StepModelValidationMap)),
```

Comment:
use PropTypes.shape over PropTypes.exact
```suggestion
  steps: PropTypes.arrayOf(PropTypes.shape(StepModelValidationMap)),
```